perfkitbenchmarker/pkb.py

```diff
@@ -320,9 +321,17 @@ def RunBenchmarks(publish=True):
 
   if FLAGS.run_uri is None:
     if FLAGS.run_stage not in [STAGE_ALL, STAGE_PREPARE]:
-      logging.error(
-          'Cannot run "%s" with unspecified run_uri.', FLAGS.run_stage)
-      return 1
+      # Attempt to get the last modified run directory.
+      run_uri = vm_util.GetLastRunUri()
+      if run_uri:
+        FLAGS.run_uri = run_uri
```

Comment:
This code only gets executed if FLAGS.run_uri is None (see line 322), which means they didn't set it.